src/heap.rs

```diff
@@ -93,6 +150,7 @@ impl HeapInterface for Heap {
 #[derive(Debug, Clone)]
 pub struct Heaps {
     heaps: Vec<Heap>,
+    recycled_pages: Vec<HeapPage>,
```

Comment:
It is possible to eliminate this Vec entirely by reusing the unused memory as a linked list.
```suggestion
    recycled_pages: Option<Box<UnusedHeapPage>>,
```
where UnusedHeapPage is something like
```rust
struct UnusedHeapPage {
    next: Option<Box<UnusedHeapPage>>,
    extra_data: [u8; HEAP_PAGE_SIZE - 8],
}
```